src/libraries/Native/Unix/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -2,14 +2,10 @@ project(System.IO.Compression.Native C)
 
 if (CLR_CMAKE_TARGET_BROWSER)
     add_definitions(-s USE_ZLIB)
-elseif (CLR_CMAKE_TARGET_ANDROID)
-    # need special case here since we want to link against libz.so but find_package() would resolve libz.a
-    set(ZLIB_LIBRARIES z)
-elseif (CLR_CMAKE_TARGET_SUNOS)
-    set(ZLIB_LIBRARIES z m)
-else ()
-    find_package(ZLIB REQUIRED)
-endif ()
+endif()
+
+set(NATIVE_LIBS_EXTRA)
+include(${CMAKE_CURRENT_LIST_DIR}/extra_libs.cmake)
```

Comment:
I don't like using ambient variables that gets set by the included code, as it is not clear from looking at this code here that the script included actually sets the variable. It would be nicer to define a function with the output variable name as a parameter in the extra_libs.cmake and call it from here after including the cmake script.
For this library, it would look like this (I hope I got the final assignment syntax right):
```cmake
function(append_extra_compression_libs NativeLibsExtra)
  if (CLR_CMAKE_TARGET_BROWSER)
      # nothing special to link
  elseif (CLR_CMAKE_TARGET_ANDROID)
      # need special case here since we want to link against libz.so but find_package() would resolve libz.a
      set(ZLIB_LIBRARIES z)
  elseif (CLR_CMAKE_TARGET_SUNOS)
      set(ZLIB_LIBRARIES z m)
  else ()
      find_package(ZLIB REQUIRED)
  endif ()
  set(${NativeLibsExtra} ${${NativeLibsExtra}} ${ZLIB_LIBRARIES} PARENT_SCOPE)
endfunction(append_extra_compression_libs)
```

Reply:
I just need to append to a list in the current cases, but other libs may need to set more information. I guess we will figure how to do that when/if we get to such needs.

I will try something as you suggest.

Reply:
Changed these to use macros.